### ISSUE TITLE
StorageJoin vs IKeyValueEntity

### ISSUE THREAD

**Issue by canhld94** [ClickHouse/ClickHouse#52188]
I think `StorageJoin` should be an `IKeyValueEntity`. Currently, to get keys from `StorageJoin` we use `joinGet`, which has a lot of common with `getByKeys`.

I'm working on a related internal features and happy to finish the code if needed. cc @vdimir

**Comment by vdimir** [comment]
@canhld94 yes, you are right, `StorageJoin` can implement this interface and behave as join with dictionary does.
